// Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";
package baidu.paddle_serving.configure;

message Shape { repeated int32 shape = 1; };

message FeedVar {
  required string name = 1;
  required bool is_lod_tensor = 2;
  required int32 feed_type = 3;
  required Shape feed_shape = 4;
}
message FetchVar {
  required string name = 1;
  required Shape fetch_shape = 2;
}
message GeneralModelConfig {
  repeated FeedVar feed_var = 1;
  repeated FetchVar fetch_var = 2;
};
